packages/imperative/src/rest/src/session/ConnectionPropsForSessCfg.ts

```diff
@@ -144,8 +144,11 @@ export class ConnectionPropsForSessCfg {
             promptForValues.push("port");
         }
 
-        if (ConnectionPropsForSessCfg.propHasValue(sessCfgToUse.tokenValue) === false &&
-            ConnectionPropsForSessCfg.propHasValue(sessCfgToUse.cert) === false) {
+        const isTokenIrrelevant = ConnectionPropsForSessCfg.propHasValue(sessCfgToUse.tokenValue) === false ||
+            (connOpts.supportedAuthTypes && !connOpts.supportedAuthTypes.includes("token"));
```

Comment:
Would it be practical to change these strings to their defined constants in SessConstants.ts, like AUTH_TYPE_TOKEN? That way, if we choose to change things in the future, it can be in one place.